modtools/create-unit.lua

```diff
@@ -631,6 +682,88 @@ function nameUnit(unit, entityRawName)
   end
 end
 
+function getItemTypeFromStr(itemTypeStr)
+--returns itemType and subType when passed a string like "ITEM_WEAPON_CROSSBOW" or "BOULDER"
+
+  local itemType = df.item_type[itemTypeStr]
+  if itemType then
+    return itemType, -1
+  end
+
+  for i = 0, df.item_type._last_item do
+    local subTypeCount = dfhack.items.getSubtypeCount(i)
+    if subTypeCount ~= -1 then
+      for subType = 0, subTypeCount-1 do
+        if dfhack.items.getSubtypeDef(i, subType).id == itemTypeStr then
+          return i, subType
+        end
+      end
+    end
+  end
+  qerror("Invalid item type: " .. itemTypeStr)
+end
+
+function extractEquipmentDetails(equipmentStr)
```

Comment:
Is this the part you were talking about potentially cleaning up?
If anything, I would try to be consistent with createitem and modtools/create-item (it looks like the latter uses `dfhack.items.createItem()`). If there's a way to make certain items more convenient for people familiar with the raws, we ought to do it everywhere.

Reply:
Yep, this is the part I was referring to. The code worked well enough during testing; I was just wondering whether there might be a better way to do this. `dfhack.items.createItem()` in itself isn't particularly useful here as the arena handles item creation for us; we just need to provide the desired item type, material and quantity. `modtools/create-item` obtains these from the user through multiple separate args, but it was written with the purpose of creating a single item per run. Here, I thought it would make more sense for everything to be specified in a single arg to make it easier to create units equipped with several items, such as an entire set of armour, as otherwise the modder would have to jump back and forth between the different args (making editing a nightmare). I tried to make the input of this new arg resemble the format used in raws, but I wouldn't recommend doing this everywhere as extracting the individual components from the input string adds an extra layer of complexity.

Reply:
Yeah, I guess I was thinking of the type/material parsing from the other plugins, not using `createItem()` itself. Makes sense why you'd want a single argument per item here, and I don't really have a good suggestion for reducing complexity. Feel free to merge when you think it's ready.